### ISSUE TITLE
Migrate jackc/pgtype CIDR to its pgx/v5/pgtype alternative

### ISSUE THREAD

**Issue by antonprokopovich** [jackc/pgx#1470]
I'm trying to migrate from `pgx/v4` to `pgx/v5` and in particular I need to rewrite code that uses `CIDR` type from `jackc/pgtype` to it's v5 alternative. I see `v5/pgtype` has the `InetCodec` type but I can't figure out how to use it instead of CIDR from v4.

Here the code I need to rewrite:

```
package main

import (
	"net"

	pgtypeV4 "github.com/jackc/pgtype" // TODO: Migrate to "github.com/jackc/pgx/v5/pgtype"
)

func toPgCIDRs(a []*net.IPNet) pgtypeV4.CIDRArray {
	elements := make([]pgtypeV4.CIDR, len(a))
	for i, e := range a {
		elements[i] = pgtypeV4.CIDR(pgtypeV4.Inet{IPNet: e, Status: pgtypeV4.Present})
	}

	return pgtypeV4.CIDRArray{
		Status:     pgtypeV4.Present,
		Elements:   elements,
		Dimensions: []pgtypeV4.ArrayDimension{{Length: int32(len(a)), LowerBound: 0}},
	}
}
```

Any kind of help would be greatly appreciated!

**Comment by jackc** [comment]
You should be able to use the core Go types directly now. There isn't a pgx CIDR type because it shouldn't be necessary anymore. You probably want https://pkg.go.dev/net/netip#Prefix.

**Comment by antonprokopovich** [comment]
@jackc Thank you for your reply! I’ll try it out.

May I ask just one more questing regarding pgtype V5 vs V4?

I have code that queries a `jsonb` filed from BD into a *string field of a struct. It worked fine with V4 but after migrating to V5 I get `scan row into struct fields: can't scan into dest[7]: json: cannot unmarshal object into Go value of type string`. 

I understand that it tries to umarshal the json and if I use `map[string]interface` instead of `*string` then error goes away. But I would like to keep using `*string` type as before and can’t figure out how to fix it.

**Comment by jackc** [comment]
Can you make an example program that demonstrates this?

**Comment by antonprokopovich** [comment]
@jackc I went down the source code and I see that  `**string` typed value is passed as `target` to `JSONCodec.PlanScan` instead of `*string`. So it [switches](https://github.com/jackc/pgx/blob/v5.2.0/pgtype/json.go#L78) to default `scanPlanJSONToJSONUnmarshal{}` case instead of the `scanPlanAnyToString{}` which causes the error.

Though I don't understand why double pointer is passed instead of a pointer.

**Comment by antonprokopovich** [comment]
Fix it on the SQL-query level by casting json to varchar like so: `CAST(r.operator_data AS VARCHAR)`. It looks more like a workaround since it's supposed to scan PG jsonb into Go *string  without any casting. It's probably something else in my code that breaks that functionality so I'll look into it later.

**Comment by jackc** [comment]
@antonprokopovich Glad you got it working. I think the `**string` case should have worked. Fixed in e48e7a7189354fd7d2990b8abc2c40dfe5ee4c9f.
